docker/prepare_proxy_test.sh

```diff
@@ -0,0 +1,194 @@
+#!/bin/sh
+
+set -o errexit
+set -o nounset
+set -o pipefail
+set -ex
+
+# customize for development / debug setup
+HUB=docker.io/$(whoami)
+TAG=test
+NAMESPACE=ccc
+GCLOUD_CLUSTER_NAME=c1
+GCLOUD_CLUSTER_ZONE=us-central1-a
+ISTIO_PROXY_UID=1337
+ISTIO_PROXY_PORT=80
+SERVER_PORT=${ISTIO_PROXY_PORT}
+CLIENT_PORT=81
+OTHER_PORT=82
+
+for image in app init proxy; do
+    bazel run //docker:${image}_debug
+    docker tag istio/docker:${image}_debug $HUB/$image:$TAG
+    docker push $HUB/$image:$TAG
+done
+
+function kc() {
+    kubectl -n ${NAMESPACE} "$@"
+}
+
+function die() {
+    echo "$@"
+    exit 1
+}
+
+kubectl get namespace ${NAMESPACE} || kubectl create namespace ${NAMESPACE}
+cat docker/prepare_proxy_test.yaml |
+    sed -e "s|TEMPLATE_HUB|${HUB}|" \
+	-e "s|TEMPLATE_TAG|${TAG}|" \
+	-e "s|CLIENT_PORT|${CLIENT_PORT}|" \
+	-e "s|SERVER_PORT|${SERVER_PORT}|" |
+    kc apply -f -
+
+function waitDeploymentReady {
+    name=${1}
+    while true; do
+	echo "waiting for all ${name} replicas to be up"
+	sleep 1
+	read WANT HAVE < <( \
+			    kc get deployment ${name} \
+			       -o go-template='{{.spec.replicas}} {{.status.replicas}}{{"\n"}}'
+	)
+	if [ -n "${WANT}" -a -n "${HAVE}" -a "${WANT}" == "${HAVE}" ]; then
+	    break
+	fi
+	echo "want ${WANT}, found ${HAVE}"
+    done
+}
+waitDeploymentReady client
+waitDeploymentReady server
+
+CLIENT=
+while [ "$CLIENT" = "" ]; do CLIENT=$(kc get pod -l app=client -o jsonpath='{.items[0].metadata.name}'); done
+CLIENT_IP=
+while [ "$CLIENT_IP" = "" ]; do CLIENT_IP=$(kc get pod -l app=client -o jsonpath='{.items[0].status.podIP}'); done
+SERVER=
+while [ "$SERVER" = "" ]; do SERVER=$(kc get pod -l app=server -o jsonpath='{.items[0].metadata.name}'); done
+SERVER_IP=
+while [ "$SERVER_IP" = "" ]; do SERVER_IP=$(kc get pod -l app=server -o jsonpath='{.items[0].status.podIP}'); done
+
+if false; then
+    # exclude specific external traffic (opt-out)
+    EXCLUDE_CIDR=$(host httpbin.org | cut -f4 -d' ' | sed -e 's|$|/32|' | paste -sd ",")
```

Comment:
Ah, this was for manual testing. I'll added a flag at the top of control this as part of the refactoring to remove the EXCLUDE option.